From 284af58c63c86e2d105e5bf82fd99e01a40e99b7 Mon Sep 17 00:00:00 2001
From: Jordan Hendricks <jordan.hendricks@joyent.com>
Date: Fri, 12 Jan 2018 23:37:24 +0000
Subject: [PATCH] MANTA-3480 Change default MPU prefix directory length to 3
 Reviewed by: Kelly McLaughlin <kelly.mclaughlin@joyent.com> Approved by: Tim
 Kordas <tim.kordas@joyent.com> Approved by: Brittany Wald
 <brittany.wald@joyent.com>

---
 lib/uploads/common.js          | 11 ++++++-----
 sapi_manifests/muskie/template |  3 ---
 2 files changed, 6 insertions(+), 8 deletions(-)

diff --git a/lib/uploads/common.js b/lib/uploads/common.js
index a6ac81b..a4d9d46 100644
--- a/lib/uploads/common.js
+++ b/lib/uploads/common.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 var util = require('util');
@@ -228,9 +228,10 @@ var UPLOAD_DIR_REGEX = new RegExp('^/([a-zA-Z][a-zA-Z0-9_\\-\\.@%]+)/uploads/([0
  * Because uuids are randomly generated, we expect that they would be uniformly
  * distributed based on the first $PREFIX_LENGTH characters of the upload id.
  *
- * We default to a $PREFIX_LENGTH of 1 -- that is, /:account/uploads will have
- * a maximum of 16 directories. If needed, this value can be tuned in SAPI to
- * allow for more ongoing multipart uploads.
+ * We default to a $PREFIX_LENGTH of 3 -- that is, /:account/uploads will have
+ * a maximum of 4096 directories. If needed, this value can be tuned in SAPI to
+ * allow for a larger or smaller number of ongoing multipart uploads per
+ * account.
  *
  * There are tradeoffs to having a higher prefix length value. A smaller prefix
  * allows all multipart uploads to be enumerated faster, as it requires fewer
@@ -240,7 +241,7 @@ var UPLOAD_DIR_REGEX = new RegExp('^/([a-zA-Z][a-zA-Z0-9_\\-\\.@%]+)/uploads/([0
  * We cap the prefix length at 4, as this is the highest prefix length that will
  * allow for under a million entries in the /:account/uploads directory.
  */
-var DEF_PREFIX_LEN = 1;
+var DEF_PREFIX_LEN = 3;
 var MIN_PREFIX_LEN = 1;
 var MAX_PREFIX_LEN = 4;
 
diff --git a/sapi_manifests/muskie/template b/sapi_manifests/muskie/template
index 56ec6d4..3393e5b 100644
--- a/sapi_manifests/muskie/template
+++ b/sapi_manifests/muskie/template
@@ -161,9 +161,6 @@
   {{#MUSKIE_MPU_PREFIX_DIR_LEN}}
     "prefixDirLen": {{MUSKIE_MPU_PREFIX_DIR_LEN}}
     {{/MUSKIE_MPU_PREFIX_DIR_LEN}}
-    {{^MUSKIE_MPU_PREFIX_DIR_LEN}}
-    "prefixDirLen": 1
-    {{/MUSKIE_MPU_PREFIX_DIR_LEN}}
   },
   {{/MPU_ENABLE}}
 
-- 
2.21.0

